File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    answer = ''
    mode = 0
    
    for i in range (0,len(code)):
        if code[i] == "1":
            mode = 0 if mode == 1 else 1   
        else:
            if (i %2 == 0 and mode == 0) or(i %2 == 1 and mode == 1):
                answer+=(code[i])
    if len(answer) == 0:
        answer = "EMPTY"
    
    return answer

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.kt
class Solution {
    fun solution(code: String): String {
        var answer: String = ""
        var mode = 0
        
        for (i in code.indices){
            if(code[i] == '1'){
                mode = if(mode == 1) 0 else 1
            }
            else{
                if((i%2==0 && mode == 0 )|| (i%2==1 && mode == 1)){
                    answer += code[i]
                }
            }
        }
        
        return if (answer.isEmpty()) "EMPTY" else answer

    }
}